presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -62,6 +62,7 @@
 
 import static com.facebook.presto.SystemSessionProperties.getTaskConcurrency;
```

Comment:
commit message is misleading as it states that `LocalExchange` is added between `TableScan` and `LookupJoin`. 
`TS` is not a side here. `LocalExchange` is just added below `LookupJoin`.

Reply:
LJO can be either in SOURCE-distributed driver or behind some partitioning. In the latter case, the number of LJO drivers is fixed already. For SOURCE-distributed LJOs don't run concurrently (that is: they are, but not all at once), so they cannot synchronize when finishing. For spill, however, they need synchronize when finishing, because unspilling is orchestrated: one spilled HBO partition is unspilled and all LJOs unspill matching data and process. When all are finished with the partition, next one is unspilled.

Thus, when spill is enabled, we need two things:
* LJOs need to run in parallel (SOURCE-distribution is no go)
** once this requirement is satisfied, there is fixed number of LJO drivers
* we need to know number of LJOs to coordinate unspilling